src/poetry/installation/installer.py

```diff
@@ -245,12 +245,9 @@ def _do_install(self) -> int:
             locked_repository = self._locker.locked_repository()
 
             if not self._locker.is_fresh():
-                self._io.write_error_line(
-                    "<warning>"
-                    "Warning: poetry.lock is not consistent with pyproject.toml. "
-                    "You may be getting improper dependencies. "
+                raise ValueError(
+                    "poetry.lock is not consistent with pyproject.toml. "
```

Comment:
@KotlinIsland What's your opinion on @ilyagr 's proposal?
Change the message to `"Dependencies changed in pyproject.toml since poetry.lock was generated. "` or something similar or keep as is?

Strictly speaking, it's not only dependencies (see https://github.com/python-poetry/poetry/blob/9756b57995bd5cf5e0b7f89bd08a7c71ae6c7d98/src/poetry/packages/locker.py#L54-L60), so `"Relevant fields"` might be more correct although all fields are related to dependencies...

Reply:
I'd say that "relevant fields" is not verbose enough for end users. If we were to use that, I think those fields would have to be named in the error message.

Reply:
And "Dependencies"? Not accurate enough or sufficient? (I wonder if it will be a bit cumbersome to name all fields?)

Reply:
I think the current message is sufficient. If it's desired to change it, perhaps that could be done in a follow-up PR.

I'd rather merge it sooner than later, and the wording can be changed in a follow-up PR if needed.

Reply:
I have no objection to doing this in a follow-up PR, but here's another thought I had:

> `pyproject.toml` changed significantly since `poetry.lock` was generated